### PR TITLE
feat:config read,update api added

### PR DESCRIPTION
**Pull request by borkarsaish65**

### CODE REVIEW

File: controllers/v1/configurationsController.js [hunk 1]
```diff
@@ -92,4 +94,102 @@ module.exports = class Configurations extends Abstract {
...
+  */
+
+	/**
+	 * Reading configuration
+	 * @method
+	 * @name read
+	 * @returns {Object} success/failure message.
+	 */
+
+	async read(req) {
```
Comment (VISHNUDAS-tunerlabs) [inline controllers/v1/configurationsController.js:131]: @borkarsaish65 add configuration for new apis, if it is missing

**Review by VISHNUDAS-tunerlabs — CHANGES_REQUESTED** [review]

File: script/defaultConfiguration.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+require('dotenv').config({ path: '../.env' });
+global.config = require('../config');
+require('../config/globalVariable')();
+const ConfigurationsHelper = require(MODULES_BASE_PATH+"/configurations/helper");
+let environmentData = require('../envVariables')();
+let code = 'keysAllowedForTargeting';
+
+let profileKeys = [
+  'state',
```
Comment (VISHNUDAS-tunerlabs) [inline script/defaultConfiguration.js:17]: @borkarsaish65 lets keep it state,district,school,block,cluster,roles for now

File: script/defaultConfiguration.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+This Node.js script is designed to add default records to the configurations collection in the database,
+ specifically targeting the initial setup of configuration keys. 
+ The script checks if there are any existing records in the collection.
+  If none are found, it will insert a predefined set of keys that are used for targeting purposes. 
+  This ensures that the necessary configuration is in place for the application to function correctly.
+*/
+
+require('dotenv').config({ path: '../.env' });
+global.config = require('../config');
+require('../config/globalVariable')();
```
Comment (VISHNUDAS-tunerlabs) [inline script/defaultConfiguration.js:11]: @borkarsaish65 lets call the script addDefaultProfileConfiguration.js

**Review by VISHNUDAS-tunerlabs — CHANGES_REQUESTED** [review]
Reviewed on 15 Oct

File: script/defaultConfiguration.js (hunk 2 above)
Comment (borkarsaish65) [reply script/defaultConfiguration.js]: done

File: script/defaultConfiguration.js (hunk 3 above)
Comment (borkarsaish65) [reply script/defaultConfiguration.js]: done

File: constants/interface-routes/configs.json [hunk 4]
```diff
@@ -1312,6 +1312,19 @@
                     "packageName": "elevate-survey"
                 }
             ]
+        },
+        {
+            "sourceRoute": "/survey/v1/profile/createOrUpdate",
```
[the remark this answers is not in the document]
Comment (borkarsaish65) [reply constants/interface-routes/configs.json]: fixed now

**Review by VISHNUDAS-tunerlabs — APPROVED** [review]
Reviewed on oct 15

**Pull request merged**